### PR TITLE
RSpec Tests

### PR DESCRIPTION
**Pull request by kaydee0502**

### CODE REVIEW

File: devsnest/spec/support/factories_support.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+HEADERS = {
+    'ACCEPT' => 'application/vnd.api+json',
+    'CONTENT-TYPE' => 'application/vnd.api+json'
+}
```
Comment (vedansh) [inline devsnest/spec/support/factories_support.rb:4]: new line

Comment (kaydee0502) [reply devsnest/spec/support/factories_support.rb]: done

File: .codeclimate.yml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+plugins:
+  rubocop:
+    enabled: true
+    channel: rubocop-0-79
```
Comment (vedansh) [inline .codeclimate.yml:4]: new line

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+before_install:
+  - mysql -e 'CREATE DATABASE devsnest_test'
+
+script:
+ - cd devsnest
+ - bundle install
+ - bundle exec rake db:create
+ - bundle exec rake db:migrate
+ - bundle exec rspec
```
Comment (vedansh) [inline .travis.yml:16]: also add spec folder

File: devsnest/Gemfile [hunk 4]
```diff
@@ -31,6 +31,7 @@ group :development, :test do
   # Call 'byebug' anywhere in the code to stop execution and get a debugger console
   gem 'byebug', platforms: %i[mri mingw x64_mingw]
   gem 'rspec-rails', '~> 3.8'
+  gem 'factory_bot_rails'
```
Comment (vedansh) [inline devsnest/Gemfile:34]: always add gem with a version number

File: devsnest/config/environments/production.rb [hunk 5]
```diff
@@ -70,16 +70,15 @@
...
-  if ENV['RAILS_LOG_TO_STDOUT'].present?
-    logger           = ActiveSupport::Logger.new($stdout)
-    logger.formatter = config.log_formatter
-    config.logger    = ActiveSupport::TaggedLogging.new(logger)
-  end
+  # if ENV['RAILS_LOG_TO_STDOUT'].present?
+  #   logger           = ActiveSupport::Logger.new($stdout)
+  #   logger.formatter = config.log_formatter
+  #   config.logger    = ActiveSupport::TaggedLogging.new(logger)
+  # end
```
Comment (vedansh) [inline devsnest/config/environments/production.rb:81]: do we need to comment this

File: devsnest/config/environments/production.rb [hunk 6]
```diff
@@ -43,7 +43,7 @@
 
   # Use the lowest log level to ensure availability of diagnostic information
   # when problems arise.
-  config.log_level = :debug
+  config.log_level = :info
```
Comment (vedansh) [inline devsnest/config/environments/production.rb:46]: what's the max level of logs, let's keep to that

File: devsnest/Gemfile [hunk 7]
```diff
@@ -60,3 +60,5 @@ gem 'tzinfo-data', platforms: %i[mingw mswin x64_mingw jruby]
 gem 'mimemagic', github: 'mimemagicrb/mimemagic', ref: '01f92d86d15d85cfd0f20dabd025dcbd36a8a60f'
 
 gem 'rubocop', '~> 1.16', require: false
+
+gem 'newrelic_rpm'
```
Comment (vedansh) [inline devsnest/Gemfile:64]: add a version

Comment (vedansh) [reply devsnest/Gemfile:64]: also add yml file to add the key for newrelic

File: devsnest/config/environments/production.rb (hunk 6 above)
Comment (kaydee0502) [reply devsnest/config/environments/production.rb:46]: there are 6 levels : `:debug`, `:info`, `:warn`, `:error`, `:fatal`, and `:unknown`

File: devsnest/config/environments/production.rb (hunk 5 above)
Comment (kaydee0502) [reply devsnest/config/environments/production.rb]: It didn't made any difference under my observation so far, uncommented.

File: devsnest/Gemfile (hunk 7 above)
Comment (kaydee0502) [reply devsnest/Gemfile:64]: No version specified from newrelic themselves, added yml file and tested by running server as production locally.

File: .travis.yml (hunk 3 above)
Comment (kaydee0502) [reply .travis.yml:16]: done

File: .codeclimate.yml (hunk 2 above)
Comment (kaydee0502) [reply .codeclimate.yml]: done

File: devsnest/config/newrelic.yml [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+# Generated June 17, 2021
+#
+# This configuration file is custom generated for NewRelic Administration
+#
+# For full documentation of agent configuration options, please refer to
+# https://docs.newrelic.com/docs/agents/ruby-agent/installation-configuration/ruby-agent-configuration
+
+common: &default_settings
+  # Required license key associated with your New Relic account.
+  license_key: d3e544a4e33722523bf0a4813cb0bf6de9f0NRAL
```
Comment (vedansh) [inline devsnest/config/newrelic.yml:15]: ENV key

**Review by vedansh — APPROVED** [review]

**Pull request merged**